File: python-pr/52/python/code-quality-error-handling/xml_parser_35.py
# {fact rule=code-quality-error-handling@v1.0 defects=0}
def parse_xml(xml_string):
    import xml.etree.ElementTree as ET
    try:
        return ET.fromstring(xml_string)
    except ET.ParseError as e:
        print(f"XML parsing error: {e}")
        return None
# {/fact}

File: js-pr/56/javascript/code-quality-error-handling/xml_parser_35.js
// {fact rule=code-quality-error-handling@v1.0 defects=0}
function parseXml(xmlString) {
    try {
        const parser = new DOMParser();
        const doc = parser.parseFromString(xmlString, 'text/xml');
        const parseError = doc.getElementsByTagName('parsererror');
        if (parseError.length > 0) {
            throw new Error('XML parsing failed');
        }
        return doc;
    } catch (error) {
        console.error('XML parsing error:', error.message);
        return null;
    }
}
// {/fact}